src/tools/runk/libcontainer/src/container.rs

```diff
@@ -69,7 +99,41 @@ impl Container {
         Ok(())
     }
 
-    // TODO: add pause and resume
+    pub fn pause(&self) -> Result<()> {
+        if self.state != ContainerState::Running && self.state != ContainerState::Created {
+            return Err(anyhow!(
+                "failed to pause container: current status is: {:?}",
+                self.state
+            ));
+        }
+        let freezer_controller: &FreezerController = self
+            .cgroup
+            .controller_of()
+            .ok_or_else(|| anyhow!("failed to get freezer controller"))?;
+        freezer_controller.freeze()?;
+        Ok(())
```

Comment:
Nit: remove `Ok(())` and  `?;` at line 113.
The same as line 129 and 135.

Reply:
line 113 and line 129 can't be simplified as above. rust will report error:

```
error[E0308]: mismatched types
   --> libcontainer/src/container.rs:114:9
    |
103 |     pub fn pause(&self) -> Result<()> {
    |                            ---------- expected `Result<(), anyhow::Error>` because of return type
...
114 |         freezer_controller.freeze()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `anyhow::Error`, found struct `cgroups_rs::error::Error`
    |
    = note: expected enum `Result<_, anyhow::Error>`
               found enum `Result<_, cgroups_rs::error::Error>`

For more information about this error, try `rustc --explain E0308`.
```